src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ConsoleDebugTargetsProvider.cs

```diff
@@ -221,29 +221,51 @@ private static bool IsRunProjectCommand(ILaunchProfile profile)
                 // If the working directory is not rooted we assume it is relative to the project directory
                 if (Path.IsPathRooted(resolvedProfile.WorkingDirectory))
                 {
-                    workingDir = resolvedProfile.WorkingDirectory;
+                    workingDir = resolvedProfile.WorkingDirectory.Replace("/", "\\");
                 }
                 else
                 {
-                    workingDir = Path.GetFullPath(Path.Combine(projectFolder, resolvedProfile.WorkingDirectory));
+                    workingDir = Path.GetFullPath(Path.Combine(projectFolder, resolvedProfile.WorkingDirectory.Replace("/", "\\")));
                 }
             }
 
             // IF the executable is not rooted, we want to make is relative to the workingDir unless is doesn't contain
-            // any path elements. In that case we are going to assume it is on the path
+            // any path elements. In that case we are going to assume it is in the current directory of the VS process, or on
+            // the environment path. If we can't find it, we just launch it as before.
             if (!string.IsNullOrWhiteSpace(executable))
             {
-                if (!Path.IsPathRooted(executable) && executable.IndexOf(Path.DirectorySeparatorChar) != -1)
+                if (!Path.IsPathRooted(executable))
```

Comment:
Just realized there's one case where this doesn't work; \Foo.exe -> should be rooted to the current drive.

Reply:
I played around with this and \foo.exe is considered rooted by Path.IsPathRooted(). Calling GetFullPath() on it does correctly prepend the current drive. I'll update the code to handle this case too..